compiler-core/src/ast/untyped.rs

```diff
@@ -31,6 +31,10 @@ pub enum UntypedExpr {
 
     // TODO: create new variant for captures specifically
     Fn {
+        /// The location from the start of the `fn` keyword to the end of the head before the opening bracket
+        head_location: SrcSpan,
```

Comment:
This duplicates the start of the `location` srcspan and so the start is redundant. It also isn't used in the code. We can store just the int index of the end instead.